model/src/w3srcemd.F90

```diff
@@ -1379,8 +1396,89 @@ SUBROUTINE W3SRCE ( srce_call, IT, JSEA, IX, IY, IMOD,          &
         END IF
 #endif
 
+#ifdef W3_PDLIB
+         IF (srce_call .eq. srce_imp_pre) THEN
+           IF (LSLOC) THEN
+             IF (IMEM == 1) THEN
+               SIDT  = PDLIB_SI(JSEA) * DTG 
+               DO IK = 1, NK
+                 JAC = CLATSL/CG1(IK) 
+                 DO ITH = 1, NTH
+                   ISP = ITH + (IK-1)*NTH
+                   VD(ISP) = MIN(0., VD(ISP))
+                   IF (LNEWLIMITER) THEN
+                     MAXDAC = MAX(DAM(ISP),DAM2(ISP))
+                   ELSE
+                     MAXDAC = DAM(ISP)
+                   ENDIF
+                   FAKS   = DTG / MAX ( 1. , (1.-DTG*VD(ISP)))
+                   DVS    = VS(ISP) * FAKS
+                   DVS    = SIGN(MIN(MAXDAC,ABS(DVS)),DVS)
+                   PreVS  = DVS / FAKS
+                   eVS    = PreVS / CG1(IK) * CLATSL
+                   eVD    = MIN(0.,VD(ISP))
+                   B_JAC(ISP,JSEA)                   = B_JAC(ISP,JSEA) + SIDT * (eVS - eVD*SPEC(ISP)*JAC)
+                   ASPAR_JAC(ISP,PDLIB_I_DIAG(JSEA)) = ASPAR_JAC(ISP,PDLIB_I_DIAG(JSEA)) - SIDT * eVD
+#endif
```

Comment:
I think this ifdef PDLIB should cover the whole "if (srce_call eq )" section, not just this first part and a few parts below.  This does not match what I recommended before and by moving the "if (srce_call eq )" inside of the w3_PDLIB section, I actually think this section of code is now even more problematic.

Reply:
Glad this was helpful! The code looks much cleaner now!